js/boaexchange.js

```diff
@@ -0,0 +1,1163 @@
+                        'orders/{orderId}',
+                        'trades',
+                        'trades/{tradeId}',
+                        'trades/all',
+                        'transactions',
+                        'transfers',
+                        'transfers/{transferId}',
+                        'withdraws',
+                    ],
+                    'post': [
+                        'addresses/{label}',
+                        'airdrops/{airdropId}',
+                        'apiKeys',
+                        'apiKeys/{key}',
+                        'chat',
+                        'markets/favorites',
+                        'markets/{label}/toggle',
+                        'orders',
+                        'transfers',
+                        'withdraws',
+                    ],
+                    'delete': [
+                        'apiKeys',
+                        'orders',
+                        'orders/{orderId}',
+                        'withdraws/{withdrawId}',
+                    ],
+                },
+            },
+            'requiredCredentials': {
+                'apiKey': true,
+                'secret': true,
+            },
+            'dontGetUsedBalanceFromStaleCache': true,
+            'fees': {
+                'trading': {
+                    'tierBased': false,
+                    'percentage': true,
+                    'maker': 0.15,
+                    'taker': 0.15,
+                },
+                'funding': {
+                    'tierBased': false,
+                    'percentage': false,
+                    'withdraw': {
+                    },
+                    'deposit': {
+                    },
+                },
+            },
+            'exceptions': {
+                'Invalid X-BOA-ENCRYPTED': AuthenticationError,
+                'Invalid X-BOA-NONCE': InvalidNonce,
+                'Address is Pending': AddressPending,
+                'Insufficient Funds': InsufficientFunds,
+                'Order Not Found': OrderNotFound,
+                'Too Many Requests': DDoSProtection,
+            },
+            'options': {
+                'hasEmptyMarkets': true,
+                // price precision by quote currency code
+                'pricePrecisionByCode': {
+                    'USD': 3,
+                },
+                'symbolSeparator': '_',
+                'tag': {
+                },
+            },
+            'commonCurrencies': {
+            },
+        });
+    }
+
+    costToPrecision (symbol, cost) {
+        return this.decimalToPrecision (cost, TRUNCATE, this.markets[symbol]['precision']['price'], DECIMAL_PLACES);
+    }
+
+    feeToPrecision (symbol, fee) {
+        return this.decimalToPrecision (fee, TRUNCATE, this.markets[symbol]['precision']['price'], DECIMAL_PLACES);
+    }
+
+    nonce () {
+        return this.seconds ();
+    }
+
+    async createDepositAddress (code, params = {}) {
```

Comment:
Is this a `createDepositAddress` or `fetchDepositAddress` ? What happens if you call it twice with the same code ? Do you get the same address in response? If that is the case, then it should be renamed to `fetchDepositAddress`. If the address returned is always different, then we should keep the current name `createDepositAddress`.

Reply:
Always different

Reply:
Ok, then the method name is correct, thanks! Do they have a method for fetching the old deposit address, or is it the policy of the exchange to never reuse them and always require to generate a new address for each new deposit?

Reply:
fetchDepositAddress returns one old address for each currency.
createDepositAddress returns a new address for the specified currency

Reply:
Pardon, i thought `fetchDepositAddress` is missing, my bad. Thanks for clarifying this! No worries then!